asv/plugins/conda.py

```diff
@@ -116,11 +116,16 @@ def install_requirements(self):
 
         self.setup()
 
+        # Install all the dependencies with a single conda command.
+        # This ensures we get the versions requested, or an error
+        # otherwise. It's also quicker than doing it one by one.
+        args = ['install', '-p', self._path, '--yes']
         for key, val in six.iteritems(self._requirements):
             if val is not None:
-                self.upgrade("{0}={1}".format(key, val))
```

Comment:
This seems to render the `upgrade` method unused. Since it's not used outside the class (and the same applies for the virtualenv variant) perhaps it should be removed from the abstract `Environment` class too? `Virtualenv.upgrade` could be made a private method.

I have a branch on stand-by with this simplification if you're interested.